components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -343,6 +350,18 @@ private String replaceInvalidChars(String username) {
         return username.replaceAll("@", "_AT_");
     }
 
+    private String createRegexPattern(List<String> redirectURIs) {
+        String regex = "";
+        for (String redirectURI : redirectURIs) {
+            if (StringUtils.isNotEmpty(regex)) {
+                regex = regex + "|" + redirectURI;
```

Comment:
This creates a final regex including the empty String. eg: (|redirect1|reidrect2). Shall we improve the logic to remove the first empty String.